runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -51,11 +51,11 @@ public final class GrpcCommitWorkStream
   private static final long HEARTBEAT_REQUEST_ID = Long.MAX_VALUE;
 
   private final Map<Long, PendingRequest> pending;
-  private final Batcher batcher;
   private final AtomicLong idGenerator;
   private final JobHeader jobHeader;
   private final ThrottleTimer commitWorkThrottleTimer;
   private final int streamingRpcBatchLimit;
+  private Batcher batcher = new Batcher();
```

Comment:
can this be final

Reply:
Oops, actually this was leftover from testing that my unit tests would catch the lack of thread-safety.  We should be creating a new batcher for each call instead of sharing the non-thread-safe batcher (similar to existing code)